lib/trip_dispatcher.rb

```diff
@@ -90,6 +91,97 @@ def load_trips
       trips
     end
 
+    def self.split_date_time(datetime)
+      #puts datetime.split("T")
+      puts Time.parse(datetime)
+    end
+
+    def available_drivers
+      available_drivers = []
+      @drivers.each do |driver|
+        if driver.status == :AVAILABLE
+          available_drivers << driver
+        end
+      end
+      return available_drivers
+    end
+
+    def driver_longest_not_driving
```

Comment:
This method is crying out for Enumerables.  That would both make it easier to read and more compact.  

It could also use being broken up into helper methods if you don't use Enumerables.